pkg/reconciler/route/traffic/traffic.go

```diff
@@ -44,6 +45,7 @@ type RevisionTarget struct {
 	Active      bool
 	Protocol    net.ProtocolType
 	ServiceName string // Revision service name.
+	Timeout     *time.Duration
```

Comment:
If duration of 0 is invalid, it is probably preferable to use `time.Duration`.
That would also simplify the code above as well.

Reply:
Sure, I can make 0 invalid.

Reply:
IIRC, that was based on RevisionSpec.TimeoutSeconds being an *int64.

Reply:
There it can be unspecified to be defaulted. Here I presume it must be specified to a finite value already?

Reply:
This gets initialized from there. Since the context is not available at that point (see traffic.go changes), I'm putting the defaulting logic in ingress.go, where that's available, and using time.Duration(0) as a magic value. Where's the best place to put the magic value?

Reply:
ingress_defaults.go?